impl/client/src/main/javascript/web/util/_a11y.js

```diff
@@ -458,6 +458,170 @@ define(["./_focus"], function(focusUtil) {
+      setFocusToFirst();
+    }
+
+    function isDisabled(item) {
+      var items = Array.from(item.classList);
+      // for cases when class is 'disabled' or 'pentaho-imagebutton-disabled'
+      return items.some(function (child) {
+        return child.includes('disabled');
+      });
+    }
+
+    function isHidden(item) {
+      return item.classList.contains('hidden');
+    }
+
+    function isNotFocusable(item) {
+      return isDisabled(item) || isHidden(item);
+    }
+
+    function setFocusItem(item) {
+      for (var i = 0; i < toolbarItems.length; i++) {
+        toolbarItems[i].setAttribute('tabindex', '-1');
+      }
+
+      if (isDisabled(item)) { //for the case when firstitem is disabled
+        item = getNext(item);
+      }
+
+      item.setAttribute('tabindex', '0');
+      item.focus();
+    }
+
+    function getNext(currentItem) {
+      var index, newItem, length = toolbarItems.length;
+
+      index = toolbarItems.indexOf(currentItem);
+      if (index < 0 || index === (length - 1)) {
+        newItem = toolbarItems[0];
+      } else {
+        newItem = toolbarItems[index + 1];
+      }
+
+      if (isNotFocusable(newItem)) {
+        return getNext(newItem);
+      }
+      return newItem;
+    }
+
+    function getPrevious(currentItem) {
+      var index, newItem, length = toolbarItems.length;
+
+      index = toolbarItems.indexOf(currentItem);
+      if (index <= 0) {
+        newItem = toolbarItems[length - 1];
+      } else {
+        newItem = toolbarItems[index - 1];
+      }
+
+      if (isNotFocusable(newItem)) {
+        return getPrevious(newItem);
+      }
+      return newItem;
+    }
+
+    function setFocusToFirst() {
+      setFocusItem(firstItem);
+    }
+
+    function setFocusToLast() {
+      setFocusItem(lastItem);
+    }
+
+    /**
+     * Handle various keyboard controls; UP/DOWN will shift focus; SPACE selects an item.
+     *
+     * @param {KeyboardEvent} evt - The keydown event object.
+     */
+    function onKeyPress(evt) {
+      var key = evt.key;
+      var flag = false;
+      var currentItem = evt.target;
+      var elemToFocus = null;
+
+      if (!currentItem) {
+        return;
+      }
+
+      switch (key) {
+
+        case keys.ArrowRight:
+          elemToFocus = getNext(currentItem);
+          setFocusItem(elemToFocus);
+          flag = true;
+          break;
+
+        case keys.ArrowLeft:
+          elemToFocus = getPrevious(currentItem);
+          setFocusItem(elemToFocus);
+          flag = true;
+          break;
+
+        case keys.Home:
+          setFocusToFirst();
+          flag = true;
+          break;
+
+        case keys.End:
+          setFocusToLast();
+          flag = true;
+          break;
+
+        case keys.Tab:
+          evt.shiftKey ? focusUtil.previousTabbable(currentItem).focus() : focusUtil.nextTabbable(currentItem).focus();
+          flag = true;
+          break;
+
+        default:
+          break;
+      }
+
+      if (flag) {
```

Comment:
Can this variable please be named after what it controls (e.g. `handled`)?

Or, ditch it, and `return`on the `default` case.